File: 004/Python/Problem004Tests.py
#!/usr/bin/env python

import unittest

import problem004

class Problem004Tests(unittest.TestCase):

    def test_is_palindrome(self):
      is_palindrome = problem004.is_palindrome
      self.assertTrue(is_palindrome(9))
      self.assertTrue(is_palindrome(33))
      self.assertTrue(is_palindrome(101))
      self.assertTrue(is_palindrome(127848721))
      self.assertTrue(is_palindrome(23233232))

    def test_is_not_palindrome(self):
      is_palindrome = problem004.is_palindrome
      self.assertFalse(is_palindrome(93))
      self.assertFalse(is_palindrome(331))
      self.assertFalse(is_palindrome(1019))
      self.assertFalse(is_palindrome(12748721))
      self.assertFalse(is_palindrome(23232323))
      
    def test_find_palindrome_product(self):
        self.assertEquals(9009, problem004.find_palindrome_product(10))
        

unittest.main()

File: 004/Erlang/problem004_tests.erl
-module(problem004_tests).

-include_lib("eunit/include/eunit.hrl").
-import(problem004, [findPalindromeProduct/1, isPalindrome/1]).

isPalindrome_test_() -> [
  ?_assert(isPalindrome(2)),
  ?_assert(isPalindrome(11)),
  ?_assert(isPalindrome(878)),
  ?_assert(isPalindrome(1134554311)),
  ?_assert(isPalindrome(9876789)),

  ?_assertNot(isPalindrome(21)),
  ?_assertNot(isPalindrome(116)),
  ?_assertNot(isPalindrome(8728)),
  ?_assertNot(isPalindrome(1134511345)),
  ?_assertNot(isPalindrome(98767829))
	].

findPalindromeProduct_test_() -> [
  ?_assertEqual(9009, findPalindromeProduct(10))
  ].
